MLResurrection.cpp

```diff
@@ -0,0 +1,668 @@
+    flSimMetadata.add_section("flsimulate.services", "flsimulate::section");
+  services_props.set_description("Services configuration");
+
+  if (!flSimParameters.servicesSubsystemConfigUrn.empty()) {
+    services_props.store_string("configUrn",
+                                flSimParameters.servicesSubsystemConfigUrn,
+                                "Services setup configuration URN");
+  } else if (!flSimParameters.servicesSubsystemConfig.empty()) {
+    services_props.store_path("config", flSimParameters.servicesSubsystemConfig,
+                              "Services setup configuration path");
+  }
+
+  return code;
+}
+
+
+void dead_cells_service(int dc[2500][3]) {
+
+  // Create a vector of `dead_cells` with dimension `N_dead_cells`
+  try {
+    std::ifstream infile(dead_cells_list);   // File for list of dead cells
+    Long64_t i=0;
+    int side, layer, column;
+    while (infile >> side >> layer >> column){
+      dead_cells[i][0]=side;
+      dead_cells[i][1]=layer;
+      dead_cells[i][2]=column;
+      i++;
+    }
+    N_dead_cells=i;
+    std::cout << "Read " << N_dead_cells << " dead cells" << std::endl;
+  }
+  catch (std::logic_error& e) {
+    std::cerr << "Problem in the list of dead cells " << e.what() << std::endl;
+  };
+
+};
+
+void write_hit(Long64_t index){
+
+  // Write a particular hit on the list of dead cells into the output calibrated data
+
+  double wx, wy; // x,y global coordinate in the detector
+  double rad = 11.0; // [mm] half radius of a cell
+  double rad_err = 11.0;  // [mm] uncertainty
+  int id; // counter for the hits in each event
+
+  // create the calibrated tracker hit to build:
+  snemo::datamodel::calibrated_data::tracker_hit_handle_type new_hit_handle(new snemo::datamodel::calibrated_tracker_hit);
+  snemo::datamodel::calibrated_tracker_hit& new_calibrated_tracker_hit = new_hit_handle.grab();
+
+  if (dead_cells[index][0]) { // Italian side
+    wx = dead_cells[index][1]*d+offsetx;
+    wy = dead_cells[index][2]*d+offsety;
+  }
+  else {                      // French side
+    wx = -(dead_cells[index][1]*d+offsetx);
+    wy = dead_cells[index][2]*d+offsety;
+  }
+
+  const geomtools::geom_id& gid = geomtools::geom_id(1204, 0, dead_cells[index][0], dead_cells[index][1], dead_cells[index][2]); // 1204 is the type of a geiger cell core
+
+  if (!calData.calibrated_tracker_hits().empty()) // get the last hit_id from the existing hits
+    id = calData.calibrated_tracker_hits().back()->get_hit_id()+index+1;
+
+  new_calibrated_tracker_hit.set_r(rad);
+  new_calibrated_tracker_hit.set_sigma_r(rad_err);
+  new_calibrated_tracker_hit.set_z(wz);
+  new_calibrated_tracker_hit.set_sigma_z(wz_err);
+  new_calibrated_tracker_hit.set_xy(wx, wy);
+  new_calibrated_tracker_hit.set_delayed(false);
+  new_calibrated_tracker_hit.set_geom_id(gid);
+  new_calibrated_tracker_hit.set_hit_id(id);
+
+  // write hit
+  new_calibrated_data->calibrated_tracker_hits().push_back(new_hit_handle);
+  hits++;
+
+}
```

Comment:
These should be factored into a class/struct with relevant data and member functions. It can form the basis of the dead cells service. It should be in a struct because `dead_cells_service` and `write_hit` are part of the service, and associated with the global data earlier.